src/api.rs

```diff
@@ -57,6 +60,44 @@ impl Api {
         Ok(Json(plan))
     }
 
+    /// Pause a plan
+    #[oai(path = "/plan/pause/:dataset_id", method = "put", tag = Tag::Plan)]
+    async fn plan_pause_put(
+        &self,
+        Data(pool): Data<&PgPool>,
+        Path(dataset_id): Path<DatasetId>,
+    ) -> poem::Result<Json<Plan>> {
+        // Start Transaction
+        let mut tx = pool.begin().await.map_err(InternalServerError)?;
+
+        // Pause a Plan
+        let plan: Plan = plan_pause_edit(&mut tx, dataset_id, true, "placeholder_user").await?;
+
+        // Commit transaction
+        tx.commit().await.map_err(InternalServerError)?;
+
+        Ok(Json(plan))
+    }
+
+    /// Unpause a plan
+    #[oai(path = "/plan/unpause/:dataset_id", method = "put", tag = Tag::Plan)]
+    async fn plan_unpause_put(
```

Comment:
@coderabbitai please create unit test for this endpoint where we set to unpause for an paused plan.